provider/ansible.rb

```diff
@@ -235,66 +234,76 @@ def get_example(cfg_file)
-        #         "examples/ansible/#{prod_name}.yaml"].join('/')
-
-        # return unless data.object.has_tests
-        # # Unlike other providers, all resources will not be built at once or
-        # # in close timing to each other (due to external PRs).
-        # # This means that examples might not be built out for every resource
-        # # in a GCP product.
-        # return unless File.file?(path)
-
-        # target_folder = data.output_folder
-
-        # name = module_name(data.object)
-        # path = File.join(target_folder,
-        #                  "test/integration/targets/#{name}/tasks/main.yml")
-        # data.generate(
-        #   'templates/ansible/integration_test.erb',
-        #   path,
-        #   self
-        # )
-
-        # # Generate 'defaults' file that contains variables.
-        # path = File.join(target_folder,
-        #                  "test/integration/targets/#{name}/defaults/main.yml")
-        # data.generate(
-        #   'templates/ansible/integration_test_variables.erb',
-        #   path,
-        #   self
-        # )
+        # Multi Cloud Swith
+        if $cloud == 'MAC'
+          azure_generate_resource_tests data
+        else
+          prod_name = data.object.name.underscore
+          path = ["products/#{data.product.api_name}",
+                  "examples/ansible/#{prod_name}.yaml"].join('/')
+
+          return unless data.object.has_tests
+          # Unlike other providers, all resources will not be built at once or
+          # in close timing to each other (due to external PRs).
+          # This means that examples might not be built out for every resource
+          # in a GCP product.
+          return unless File.file?(path)
+
+          target_folder = data.output_folder
+
+          name = module_name(data.object)
+          path = File.join(target_folder,
+                           "test/integration/targets/#{name}/tasks/main.yml")
+          data.generate(
+            'templates/ansible/integration_test.erb',
+            path,
+            self
+          )
+
+          # Generate 'defaults' file that contains variables.
+          path = File.join(target_folder,
+                           "test/integration/targets/#{name}/defaults/main.yml")
+          data.generate(
+            'templates/ansible/integration_test_variables.erb',
+            path,
+            self
+          )
+        end
+      end
+
+      def gcp_compile_datasource(data)
+        
       end
 
       def compile_datasource(data)
-        # TODO: Azure switch
-        azure_compile_datasource(data)
-
-        # target_folder = data.output_folder
-        # name = "#{module_name(data.object)}_facts"
-        # data.generate('templates/ansible/facts.erb',
-        #               File.join(target_folder,
-        #                         "lib/ansible/modules/cloud/google/#{name}.py"),
-        #               self)
+        # Multi Cloud Swith
+        if $cloud == 'MAC'
```

Comment:
Similar comment as before. We might be able to minimize the changes to one line.

Reply:
Resolved.